### PR TITLE
Added BytesIO support to PipelineBase.load()

### LINKED ISSUE #4178: Update PipelineBase.load() to accept both a path and a file-like object
Currently, PipelineBase.load() accepts just a file path.  I have some scenarios in which I have a BytesIO object that I would like to pass into the function.

Proposed update: 
1. Allow str or BytesIO as the argument
2. If BytesIO, pass directly to cloudpickle, which already handles BytesIO
3. If str, business as usual

### PR DESCRIPTION
**Pull request by cp2boston**
### Pull Request Description
Added support for BytesIO input to PipelineBase.load(), primarily to support VFS

Fixes #4178

### CODE REVIEW

**Comment by machineFL** [comment]
No linked issues found. Please add the corresponding issues in the pull request description. <br/> [Use GitHub automation to close the issue when a PR is merged](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

**Review by jeremyliweishih — APPROVED** [review]
Nit but otherwise LGTM!

File: evalml/tests/pipeline_tests/test_pipelines.py [hunk 1]
```diff
@@ -132,13 +133,27 @@ def test_serialization(X_y_binary, tmpdir, logistic_regression_binary_pipeline):
     path = os.path.join(str(tmpdir), "pipe.pkl")
     pipeline = logistic_regression_binary_pipeline
     pipeline.fit(X, y)
+
```
[the remark this answers is not in the document]
Comment (cp2boston) [reply evalml/tests/pipeline_tests/test_pipelines.py:140]: Great idea!  Done

[the remark this answers is not in the document]
Comment (cp2boston) [reply evalml/tests/pipeline_tests/test_pipelines.py:140]: evalml/tests/pipeline_tests/test_pipelines.py::test_serialization[str] PASSED                                                           [ 50%]
evalml/tests/pipeline_tests/test_pipelines.py::test_serialization[BytesIO] PASSED                                                       [100%]

**Pull request merged**